OnBoardingCalculator/app/src/main/java/com/globant/onboardingcalculator/mvp/presenter/CalculatorPresenter.java

```diff
@@ -20,18 +33,74 @@ public void onClearPressed() {
-        if (model.getOperator() == Constants.EMPTY_CHAR)
+        if (!model.getFirstOperand().isEmpty()) {
             model.setOperator(operator);
-        view.refreshVisor(String.valueOf(operator));
+            view.refreshVisor(String.valueOf(operator));
+            view.enablePointBtn();
+        } else
+            view.showOperatorError();
     }
 
     public void onNumberPressed(String number) {
-        if (model.getOperator() == Constants.EMPTY_CHAR) {
+        if (!model.getResult().equals(EMPTY_STRING) && (model.getOperator() == EMPTY_CHAR))
+            view.showOperatorErrorAfterEqualPressed();
+        else if (model.getOperator() == EMPTY_CHAR) {
             model.setFirstOperand(number);
             view.refreshVisor(model.getFirstOperand());
         } else {
             model.setSecondOperand(number);
             view.refreshVisor(model.getSecondOperand());
         }
     }
-}
\ No newline at end of file
+
+    public void onEqualPressed() {
+        if (!model.emptyOperation())
+            switch (model.getOperator()) {
+                case OPERATOR_PLUS:
+                    model.operate(String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) + parseDouble(model.getSecondOperand()))));
+                    break;
+                case OPERATOR_SUBSTRACTION:
+                    String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) - parseDouble(model.getSecondOperand())));
+                    model.operate(String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) - parseDouble(model.getSecondOperand()))));
+                    break;
+                case OPERATOR_MULTIPLY:
+                    model.operate(String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) * parseDouble(model.getSecondOperand()))));
+                    break;
+                case OPERATOR_DIVIDE:
+                    if (parseDouble(model.getSecondOperand()) != parseDouble(NUMBER_ZERO)) {
+                        model.operate(String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) / parseDouble(model.getSecondOperand()))));
+                    } else {
+                        model.clearOperation();
+                        view.showMathError();
+                        break;
+                    }
+            }
+        view.refreshVisor(model.getResult());
+    }
+
+    public void onPointPressed() {
+        if (!model.getResult().equals(EMPTY_STRING) && (model.getOperator() == EMPTY_CHAR))
+            view.showOperatorErrorAfterEqualPressed();
+        else if (model.getOperator() == EMPTY_CHAR) {
+                if (model.getFirstOperand().isEmpty()) {
+                    model.setFirstOperand(NUMBER_ZERO);
+                    model.setFirstOperand(DECIMAL_POINT);
+                    view.refreshVisor(model.getFirstOperand());
+                } else if (model.getFirstOperand().contains(DECIMAL_POINT))
```

Comment:
please use braces in this if to get more readability

Reply:
Done!